cobalt/updater/configurator.cc

```diff
@@ -0,0 +1,422 @@
+  if (!locale_id) {
+    // If Starboard failed to determine the locale, return a hardcoded, but
+    // valid BCP 47 language tag as required by
+    // https://html.spec.whatwg.org/commit-snapshots/e2f08b4e56d9a098038fb16c7ff6bb820a57cfab/#language-preferences
+    return "en-US";
+  }
+  std::string locale_string(locale_id);
+  // POSIX platforms put time zone id at the end of the locale id, like
+  // |en_US.UTF8|. We remove the time zone id.
+  int first_dot = locale_string.find_first_of(".");
+  if (first_dot != std::string::npos) {
+    return locale_string.substr(0, first_dot);
+  }
+  return locale_string;
+}
+
+std::string Configurator::GetOSLongName() const {
+  return GetDeviceProperty(kSbSystemPropertyPlatformName);
+}
+
+base::flat_map<std::string, std::string> Configurator::ExtraRequestParams()
+    const {
+  base::flat_map<std::string, std::string> params;
+  params.insert(std::make_pair("SABI", SB_SABI_JSON_ID));
+  params.insert(std::make_pair("sbversion", std::to_string(SB_API_VERSION)));
+
+  // The flag name to force an update is changed to "is_forced_update_" to cover
+  // the cases where update is requested by client but channel stays the same,
+  // but the flag name sent to Omaha remains "updaterchannelchanged" to isolate
+  // the changes to Omaha configs. If it's decided to change this flag name
+  // on Omaha, it'll be done in a separate PR after the Omaha configs are
+  // updated.
+  params.insert(std::make_pair(
+      "updaterchannelchanged",
+      std::atomic_load(&is_forced_update_) == 1 ? "True" : "False"));
+  // Brand name
+  params.insert(
+      std::make_pair("brand", GetDeviceProperty(kSbSystemPropertyBrandName)));
+
+  // Model name
+  params.insert(
+      std::make_pair("model", GetDeviceProperty(kSbSystemPropertyModelName)));
+
+  // Original Design manufacturer name
+  params.insert(
+      std::make_pair("manufacturer",
+                     GetDeviceProperty(kSbSystemPropertySystemIntegratorName)));
+
+  // Chipset model number
+  params.insert(std::make_pair(
+      "chipset", GetDeviceProperty(kSbSystemPropertyChipsetModelNumber)));
+
+  // Firmware version
+  params.insert(std::make_pair(
+      "firmware", GetDeviceProperty(kSbSystemPropertyFirmwareVersion)));
+
+  // Model year
+  params.insert(
+      std::make_pair("year", GetDeviceProperty(kSbSystemPropertyModelYear)));
+
+  // User Agent String
+  params.insert(std::make_pair("uastring", user_agent_string_));
+
+  // Certification scope
+  params.insert(std::make_pair(
+      "certscope", GetDeviceProperty(kSbSystemPropertyCertificationScope)));
+
+  // Compression status
+  params.insert(std::make_pair("usecompressedupdates",
+                               GetUseCompressedUpdates() ? "True" : "False"));
+
+  return params;
+}
+
+std::string Configurator::GetDownloadPreference() const {
+  return {};
+}
+
+scoped_refptr<update_client::NetworkFetcherFactory>
+Configurator::GetNetworkFetcherFactory() {
```

Comment:
nit: Simple accessors and mutators (getters and setters) are usually
named with snake case names in Chromium, so this would be 
`get_network_fetcher_factory()`. Same in l. 239, 244 etc. This
conveys eg that there's no lock involved.


> Accessors and mutators (get and set functions) may be named like variables, in snake_case. These often correspond to actual member variables, but this is not required. For example, int count() and void set_count(int count).